### ISSUE TITLE
Can't download nebula videos

### ISSUE THREAD

**Issue by Blackilykat** [futo-org/grayjay-android#1075]
Whenever I try to download a video from nebula, I get the error `Failed to get variant playlist: 404`
![Screenshot_20240615-013053.png](https://github.com/futo-org/grayjay-android/assets/66535696/72766798-8a19-45fd-9ea7-7c4966b6805e)

**Comment by intgr** [comment]
Indeed, download is completely broken with Nebula. The 404 error occurs with any video on Nebula, regardless of video format or quality.

I regularly listen to videos while commuting. Since Nebula only offers cooked files with audio+video together, this quickly racks up mobile data usage even with the lowest video quality, enough that I had to upgrade my data plan. (Though it's completely wasted because I don't actually watch the video). I would download on WiFi at home if it were possible.

**Comment by kaidelorenzo** [comment]
Should be addressed with this merge request. It will hopefully make it into the next Android release of Grayjay

https://gitlab.futo.org/videostreaming/grayjay/-/merge_requests/75

**Comment by Zvonimir-FUTO** [comment]
Fixed in v312.

**Comment by alebahn** [comment]
I'm still experiencing this issue on version 327.

**Comment by Zvonimir-FUTO** [comment]
> I'm still experiencing this issue on version 327.

Please open a separate issue for that since the root of original issue was fixed so this is likely something different.
